File: java/air-io/src/main/java/feuyeux/io/nio/udp/protocol/StringInputMessage.java
package feuyeux.io.nio.udp.protocol;

public class StringInputMessage extends BaseMessage {

    private String value;

    public StringInputMessage() {
        cmdType = CMD_TYPE_STRING_INPUT;
    }

    public StringInputMessage(String value) {
        cmdType = CMD_TYPE_STRING_INPUT;
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "[StringInput] TYPE=" + cmdType + ",VALUE=" + value;
    }
}

File: ios/learning/UdpCmd/UdpCmd/StringInputMessage.m
#import "StringInputMessage.h"

@implementation StringInputMessage
- (NSString *)toString {
    return [NSString localizedStringWithFormat:@"TYPE=%d,VALUE=%@", self.cmdType, _value];
}

- (StringInputMessage *)init:(NSString *)value {
    self = [super init];
    self.cmdType = STRINGS;
    [self setValue:value];
    return self;
}
@end
